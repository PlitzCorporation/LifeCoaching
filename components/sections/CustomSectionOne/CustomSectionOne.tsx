import React from 'react';
import Image from 'next/image';

import altCreator from '@/helpers/altCreator';

import BurgundyThickLine from '@/components/ui/decorations/BurgundyThickLine';

import { ExtraBoxesProps } from '@/types/pages';

export interface CustomSectionOneProps extends ExtraBoxesProps {
	testId?: string;
}

const CustomSectionOne = (props: CustomSectionOneProps) => {
	const {
		testId = 'CustomSectionOne',
		econtent,
		etitle,
		esubtitle,
		eimg,
	} = props;

	const altText = altCreator(eimg as string);

	return (
		<section data-testid={testId} className="plitz-container py-10 lg:py-20">
			<div className={`grid grid-cols-1 lg:grid-cols-12 gap-14`}>
				<div
					className={`h-full flex flex-col justify-center ${eimg && eimg.includes('placeholder') ? 'lg:col-span-12' : 'lg:col-span-7'} text-balance`}
				>
					<BurgundyThickLine />
					<p className="text-plitz-primary font-bold mt-10 mb-5">{esubtitle}</p>
					<h3 className="text-balance mb-10">{etitle}</h3>
					<div
						dangerouslySetInnerHTML={{ __html: econtent }}
						className="whitespace-pre-wrap mb-12 lg:mb-16"
					></div>
				</div>
				{eimg && !eimg.includes('placeholder') && (
					<div className="lg:col-span-5">
						<Image
							src={`${process.env.NEXT_PUBLIC_CMS}/${eimg}`}
							alt={altText}
							width={427}
							height={500}
							className="w-full max-w-96 mx-auto lg:max-w-full"
						/>
					</div>
				)}
			</div>
		</section>
	);
};

export default CustomSectionOne;
