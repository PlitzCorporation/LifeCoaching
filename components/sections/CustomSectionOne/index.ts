import CustomSectionOne from './CustomSectionOne';

export default CustomSectionOne;
