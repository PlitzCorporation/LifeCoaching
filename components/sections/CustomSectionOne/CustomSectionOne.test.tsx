import { render } from '@testing-library/react';
import CustomSectionOne from './CustomSectionOne';

jest.mock('next/image', () => ({
	__esModule: true,
	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	default: (props: any) => {
		// eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
		return <img {...props} priority="false" />;
	},
}));

describe('CustomSectionOne', () => {
	const mockData = {
		_id: '1',
		etitle: 'Welcome to Reason4Hope',
		esubtitle: 'Hi, I am Brenda Silva',
		econtent: 'This is Brenda SilvaÂ´s Bio',
		eimg: 'uploads/Reason4Hope/image.jpg',
		eposition: 3,
	};

	const OLD_ENV = process.env;

	beforeAll(() => {
		jest.resetModules();
		process.env.NEXT_PUBLIC_CMS = 'https://thedavid.plitz7.com';
	});

	afterAll(() => {
		process.env = OLD_ENV;
	});

	it('should render', () => {
		const { container } = render(<CustomSectionOne {...mockData} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
