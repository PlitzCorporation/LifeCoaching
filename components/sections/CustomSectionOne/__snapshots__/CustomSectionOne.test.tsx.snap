// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSectionOne should render 1`] = `
<section
  class="plitz-container py-10 lg:py-20"
  data-testid="CustomSectionOne"
>
  <div
    class="grid grid-cols-1 lg:grid-cols-12 gap-14"
  >
    <div
      class="h-full flex flex-col justify-center lg:col-span-7 text-balance"
    >
      <div
        class="block w-32 h-1 bg-gradient-to-r from-plitz-primary/5 to-plitz-primary/100"
      />
      <p
        class="text-plitz-primary font-bold mt-10 mb-5"
      >
        Hi, I am Brenda Silva
      </p>
      <h3
        class="text-balance mb-10"
      >
        Welcome to Reason4Hope
      </h3>
      <div
        class="whitespace-pre-wrap mb-12 lg:mb-16"
      >
        This is Brenda SilvaÂ´s Bio
      </div>
    </div>
    <div
      class="lg:col-span-5"
    >
      <img
        alt="uploads/Reason4Hope/image"
        class="w-full max-w-96 mx-auto lg:max-w-full"
        height="500"
        priority="false"
        src="https://thedavid.plitz7.com/uploads/Reason4Hope/image.jpg"
        width="427"
      />
    </div>
  </div>
</section>
`;
